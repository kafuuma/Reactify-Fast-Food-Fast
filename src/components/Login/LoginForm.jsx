import React from 'react';
import {
  MDBContainer,
  MDBRow,
  MDBCol,
  MDBInput,
  MDBBtn,
  MDBCard,
  MDBCardBody
} from 'mdbreact';

const Login = ({ handleChange, handleSubmit }) => {
  const bgPink = { background: '#ff3300' };
  return (
    <MDBContainer>
      <MDBRow className="d-flex flex-center pt-5">
        <MDBCol md="6">
          <MDBCard>
            <MDBCardBody>
              <form onSubmit={handleSubmit}>
                <p className="h4 text-center py-4">Login</p>
                <div className="grey-text">
                  <MDBInput
                    name="email"
                    label="Your email"
                    icon="envelope"
                    group
                    type="email"
                    validate
                    error="wrong"
                    success="right"
                    onChange={handleChange}
                  />
                  <MDBInput
                    name="password"
                    label="Your password"
                    icon="lock"
                    group
                    type="password"
                    validate
                    onChange={handleChange}
                  />
                </div>
                <div className="text-center py-4 mt-3">
                  <MDBBtn id="bg" type="submit">
                    Login
                  </MDBBtn>
                </div>
              </form>
            </MDBCardBody>
          </MDBCard>
        </MDBCol>
      </MDBRow>
    </MDBContainer>
  );
};

export default Login;
