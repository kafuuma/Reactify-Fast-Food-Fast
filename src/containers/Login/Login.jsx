import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LoginForm from '../../components/Login/LoginForm';
import LoginUser from '../../actions/authActions/loginAction';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      message: ''
    };
  }

  componentWillReceiveProps(nextProps) {
    const errors = nextProps.data;
    if (errors) {
      this.setState({ errors: errors });
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
    console.log(this.state);
  };

  handleSubmit = e => {
    console.log('this was deleted');
    e.preventDefault();
    const { LoginUser } = this.props;
    LoginUser(this.state);
  };

  render() {
    return (
      <LoginForm
        handleChange={this.handleChange}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

Login.propTypes = {
  LoginUser: PropTypes.func
};

Login.defaultProps = {
  LoginUser: () => {}
};

export const mapStateToProps = state => ({
  //data: state.user.user
});

export default connect(
  mapStateToProps,
  { LoginUser }
)(Login);
