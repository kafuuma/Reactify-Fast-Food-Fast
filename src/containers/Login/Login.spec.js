import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

const state = {
  email: '',
  password: ''
};
const toast = jest.fn;

describe('<Login />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  it('matches snapshort', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call handleChange ', () => {
    const instance = wrapper.instance();
    expect(
      instance.handleChange({
        preventDefault: jest.fn,
        target: {
          name: 'email'
        }
      })
    );
  });
  it('should call handleSubmit ', () => {
    const instance = wrapper.instance();
    expect(
      instance.handleSubmit({
        preventDefault: jest.fn
      })
    );
  });

  it('should handle login success ', () => {
    const message = {
      message: {
        message: 'login successful',
        Athentication: 'moked token'
      }
    };
    const props = {
      message: 'login successful',
      toast
    };
    const testWrapper = shallow(<Login />);
    testWrapper.setProps({ ...props });
    expect(testWrapper.props());
  });
  it('should handle login failure ', () => {
    const message = {
      message: {
        message: 'login failure',
        Athentication: 'moked token'
      }
    };
    const props = {
      message,
      toast
    };
    wrapper.setProps(props);
    expect(wrapper.state()).toEqual(state);
  });
});
