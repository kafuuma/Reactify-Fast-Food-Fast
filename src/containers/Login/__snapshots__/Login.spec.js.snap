// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshort 1`] = `ShallowWrapper {}`;
